### PR TITLE
Book Suggestion rspec

### PR DESCRIPTION
**Pull request by franciscoabalan**
## Summary
Rspec test to book suggestion controller

## Trellocard
https://trello.com/c/MUJ6086T/27-crear-test-para-endpoint-de-creaci%C3%B3n-de-un-booksuggestion

## Screenshot
Captura de Pantalla 2019-10-30 a la(s) 15 36 22

### CODE REVIEW

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+describe Api::V1::BookSuggestionsController do
+  describe 'POST #create' do
```
Comment (joredudiaz) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb:4]: it is good, but we need to ensure also that the record is saved. You can use `change` from rspec to see if the count of the records changed or not

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    subject(:http_request) { post :create, params: valid_attributes }
+
+    let(:valid_attributes) do
+      { user_id: user_id, synopsis: 'test_synopsis', author: 'test_author',
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]: I think that create this params with faker, could be a little better

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        title: 'test_title', link: 'test_link', publisher: 'test_publisher',
+        year: '1987', price: 5000 }
+    end
+
+    before do
+      http_request
+    end
+
+    context 'when request is valid' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
    context 'when the request is valid' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      let(:user_id) { create(:user).id }
+
+      it 'responds with 201 status' do
+        expect(http_request).to have_http_status(:created)
+      end
+
+      it 'when create the response.body valid user' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct user_id' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        expect(JSON.parse(response.body)['user_id']).to eq(valid_attributes[:user_id])
+      end
+
+      it 'when create the response.body valid author' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct author' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        expect(JSON.parse(response.body)['author']).to eq(valid_attributes[:author])
+      end
+
+      it 'when create the response.body valid title' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct title' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        expect(JSON.parse(response.body)['title']).to eq(valid_attributes[:title])
+      end
+
+      it 'when create the response.body valid link' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct link' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        expect(JSON.parse(response.body)['link']).to eq(valid_attributes[:link])
+      end
+
+      it 'when create the response.body valid publisher' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct publisher' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        expect(JSON.parse(response.body)['publisher']).to eq(valid_attributes[:publisher])
+      end
+      it 'when create the response.body valid year' do
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
      it 'returns the correct year' do
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        expect(JSON.parse(response.body)['year']).to eq(valid_attributes[:year])
+      end
+
+      it 'creates a new book suggestions' do
+        expect do
+          post :create, params: valid_attributes
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
          expect { http_request }.to change(BookSuggestion, :count)
```

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        expect(http_request.body['error']).to be_present
+      end
+
+      it 'responds with 422 status' do
+        expect(http_request).to have_http_status(:unprocessable_entity)
+      end
+
+      it 'Do not creates a new book suggestions' do
+        expect do
+          post :create, params: valid_attributes
```
Comment (aleochoam) [inline spec/controllers/api/v1/book_suggestions_controller_spec.rb]:
```suggestion
          expect { http_request }.not_to change(BookSuggestion, :count)
```

**Review by aleochoam — COMMENTED** [review]
😅

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 1 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb:4]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 2 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 3 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 4 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 5 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 6 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 7 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 8 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 9 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 11 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

File: spec/controllers/api/v1/book_suggestions_controller_spec.rb (hunk 10 above)
Comment (franciscoabalan) [reply spec/controllers/api/v1/book_suggestions_controller_spec.rb]: Done

**Review by aleochoam — APPROVED** [review]
👏

**Review by joredudiaz — APPROVED** [review]
good! 😄

**Pull request merged**